src/records/DateRecord.js

```diff
@@ -0,0 +1,6 @@
+import {Record} from 'immutable';
```

Comment:
Calling this `DateRecord` is inconsistent with the other records (which don’t explicitly have `Record` in their names). I’m guessing you did that to disambiguate with the native `Date`, which is a good idea, but maybe something like `FormDate` or `ParsedNaturalLanguageDate` would work?